Makefile

```diff
@@ -131,6 +131,16 @@ tools:
 .PHONY: run
 run: build
 	@cargo osdk run $(CARGO_OSDK_ARGS)
+# Check the running status of auto tests from the QEMU log
+ifeq ($(AUTO_TEST), syscall)
+	@tail --lines 100 qemu.log | grep -q "^.* of .* test cases passed." || (echo "Syscall test failed" && exit 1)
+endif
+ifeq ($(AUTO_TEST), regression)
+	@tail --lines 100 qemu.log | grep -q "^All regression tests passed." || (echo "Regression test failed" && exit 1)
+endif
+ifeq ($(AUTO_TEST), boot)
+	@tail --lines 100 qemu.log | grep -q "^Successfully booted." || (echo "Boot test failed" && exit 1)
+endif
```

Comment:
One more thing. In this Makefile, there are two occurrences of the following `if` pattern.

```Makefile
ifeq <cond1>
	<branch1>
endif
ifeq <cond2>
	<branch2>
endif
ifeq <cond3>
	<branch2>
endif
```

It can be revised to the following form.

```Makefile
ifeq <cond1>
	<branch1>
else ifeq <cond2>
	<branch2>
else ifeq <cond3>
	<branch3>
endif
```

The revised form is shorter and more clear (the branches are exclusive).